components/server/ee/src/workspace/gitpod-server-impl.ts

```
@@ -2738,4 +2760,84 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
+        await this.guardAdminAccess("adminGetCostCenter", { id: user.id }, Permission.ADMIN_USERS);
+
+        const { costCenter } = await this.usageService.getCostCenter({ attributionId });
+        return this.translateCostCenter(costCenter);
+    }
+
+    async adminSetUsageLimit(ctx: TraceContext, attributionId: string, usageLimit: number): Promise<void> {
+        const attrId = AttributionId.parse(attributionId);
+        if (attrId === undefined) {
+            log.error(`Invalid attribution id: ${attributionId}`);
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, `Invalid attibution id: ${attributionId}`);
+        }
+        if (typeof usageLimit !== "number" || usageLimit < 0) {
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, `Unexpected usageLimit value: ${usageLimit}`);
+        }
+        const user = this.checkAndBlockUser("adminSetUsageLimit");
+        await this.guardAdminAccess("adminSetUsageLimit", { id: user.id }, Permission.ADMIN_USERS);
+
+        const response = await this.usageService.getCostCenter({ attributionId });
+
+        // backward compatibility for cost centers that were created before introduction of BillingStrategy
+        if (!response.costCenter) {
+            throw new ResponseError(ErrorCodes.NOT_FOUND, `Coudln't find cost center with id ${attributionId}`);
+        }
+        const stripeSubscriptionId = await this.findStripeSubscriptionId(ctx, attributionId);
+        if (stripeSubscriptionId != undefined) {
+            response.costCenter.billingStrategy = CostCenter_BillingStrategy.BILLING_STRATEGY_STRIPE;
+        }
+
+        await this.usageService.setCostCenter({
+            costCenter: {
+                attributionId,
+                spendingLimit: usageLimit,
+                billingStrategy: response.costCenter.billingStrategy,
+            },
+        });
+
+        this.messageBus.notifyOnSubscriptionUpdate(ctx, attrId).catch();
```

Intentional empty catch?